hack/lib/util/golang.sh

```diff
@@ -8,10 +8,10 @@ function os::golang::verify_go_version() {
 
 	local go_version
 	go_version=($(go version))
-	if [[ "${go_version[2]}" != go1.7* ]]; then
+	if [[ "${go_version[2]}" != go1.8* ]]; then
 		os::log::info "Detected go version: ${go_version[*]}."
 		if [[ -z "${PERMISSIVE_GO:-}" ]]; then
-			os::log::error "Please install Go version 1.7 or use PERMISSIVE_GO=y to bypass this check."
+			os::log::error "Please install Go version 1.8 or use PERMISSIVE_GO=y to bypass this check."
```

Comment:
I've added revert of 715e2f68718a7db9b93916ff95de60993d9bac1b to this pull request -> 988a8dbbbaafe3b8d81a58a75999f55efe7eb762.

The original approach used `os::log::error` + `return 1` so I did stick to that.